docs/book/driver-deployment/upgrade.md

```diff
@@ -4,14 +4,82 @@
+
+vSAN file service volumes (Read Write Many vSphere CSI Volumes) become inaccessible after upgrading the CSI driver. Refer to this [issue](https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/1216) for more detail.
+This issue is present in vSphere CSI Driver v2.0.0 to v2.2.1. Fix for this issue will be provided in `v2.3.0` release. New Pods with file share volumes created using v2.3.0 will not run into this issue.
+The workaround for this issue is to remount vSAN file services volumes used by the application pods at the same location from the Node VM Guest OS directly. Please perform the following steps before upgrading the driver.
+
+**Warning:** Following steps should be performed in a maintenance window and, it might disrupt active IOs on the file share volumes used application Pods. If you have multiple replicas of the Pod accessing the same file share volume, it is recommended to perform the below steps on each mount point serially to minimize downtime and disruptions.
+
+1. Find all RWM volumes on the Cluster.
+
+    ```bash
+    # kubectl get pv -o wide | grep RWX
+    pvc-7e43d1d3-2675-438d-958d-41315f97f42e   1Gi        RWX            Delete           Bound    default/www-web-0   file-sc                 107m   Filesystem
+    ```
+
+2. Find all Nodes to which RWM volume is attached. In here below example, we can see volume is attached and mounted on the `k8s-worker3` node.
+
+    ```bash
+    # kubectl get volumeattachment | grep pvc-7e43d1d3-2675-438d-958d-41315f97f42e
+    csi-3afe670705e55e0679cba3f013c78ff9603333fdae6566745ea5f0cb9d621b20   csi.vsphere.vmware.com   pvc-7e43d1d3-2675-438d-958d-41315f97f42e   k8s-worker3   true       22s
+    ```
+
+3. Log in to the `k8s-worker3` node VM and execute the following command to know where the volume is mounted to.
+
+    ```bash
+    root@k8s-worker3:~# mount | grep pvc-7e43d1d3-2675-438d-958d-41315f97f42e
+    10.83.28.38:/52d7e15c-d282-3bae-f64d-8851ad9d352c on /var/lib/kubelet/pods/43686ba4-d765-4378-807a-74049fca39ee/volumes/kubernetes.io~csi/pvc-7e43d1d3-2675-438d-958d-41315f97f42e/mount type nfs4 (rw,relatime,vers=4.1,rsize=1048576,wsize=1048576,namlen=255,hard,proto=tcp,timeo=600,retrans=2,sec=sys,clientaddr=10.244.3.134,local_lock=none,addr=10.83.28.38)
+    ```
+
+4. unmount and remount volume at the same location with the same mount options used for mounting volume. For this step, the `nfs-common` package needs to be pre-installed on the worker VMs.
+
+   1. Umount the volume
+      - Use the `umount -f` command to unmount the volume with force.
+
+         ```bash
+         root@k8s-worker3:~# umount -f /var/lib/kubelet/pods/43686ba4-d765-4378-807a-74049fca39ee/volumes/kubernetes.io~csi/pvc-7e43d1d3-2675-438d-958d-41315f97f42e/mount
+         ```
+
+      - If the volume is busy or already inaccessible, use `umount -l` command to do the lazy unmount.
```

Comment:
Instead of 2 instructions for 2 scenarios, can we just include everything(-l option & -f option) in 1 instruction

```
root@k8s-worker3:~# umount -lf /var/lib/kubelet/pods/43686ba4-d765-4378-807a-74049fca39ee/volumes/kubernetes.io~csi/pvc-7e43d1d3-2675-438d-958d-41315f97f42e/mount
```

Reply:
Checked with Sriram, we are good with `umoint -fl`. I will make the change.